pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -75,16 +77,17 @@ def _check_consider_iterating_dictionary(self, node: nodes.Call) -> None:
             return
         if node.func.attrname != "keys":
             return
-        if not isinstance(node.parent, (nodes.For, nodes.Comprehension)):
-            return
-
-        inferred = utils.safe_infer(node.func)
-        if not isinstance(inferred, astroid.BoundMethod) or not isinstance(
-            inferred.bound, nodes.Dict
+        if (
+            isinstance(node.parent, (nodes.For, nodes.Comprehension))
+            or isinstance(node.parent, nodes.Compare)
+            and any(op for op, _ in node.parent.ops if op == "in")
```

Comment:
It's a bit constructed, but technically possible. No warning should be emitted for
```py 
var = [1, 2] == {}.keys() in {False}
```

It should be enough to add an additional condition to the `if` clause, testing that the second value of `ops` is in fact the call `node` itself.

Reply:
Some day I want to know as much about Python as you seem to do and be able to come up with these tests 😄

Reply:
Thanks for the compliment ❤️